crates/agglayer-config/src/storage.rs

```diff
@@ -104,6 +118,36 @@ impl From<StorageConfig> for StorageConfigHelper {
     }
 }
 
+fn default_db_path() -> PathBuf {
+    PathBuf::new().join(STORAGE_DIR)
+}
+
+pub(crate) fn normalize_path(path: &Path) -> PathBuf {
+    let mut components = path.components().peekable();
+    let mut ret = if let Some(c @ Component::Prefix(..)) = components.peek().cloned() {
+        components.next();
+        PathBuf::from(c.as_os_str())
+    } else {
+        PathBuf::new()
+    };
+
+    for component in components {
+        match component {
+            Component::Prefix(..) => unreachable!(),
+            Component::RootDir => {
+                ret.push(component.as_os_str());
+            }
+            Component::CurDir => {}
+            Component::ParentDir => {
+                ret.pop();
+            }
```

Comment:
What's the intended behaviour in presence of symlinks? Should `foo/..` resolve to `.` even if `foo` is a symlink pointing somewhere else? There's the `Path::canonicalize` method which follows symlinks but it requires the path to physically exist.